test/image/compare_pixels_test.js

```diff
@@ -1,109 +1,154 @@
 var fs = require('fs');
-var path = require('path');
 
-var constants = require('../../tasks/util/constants');
-var getOptions = require('../../tasks/util/get_image_request_options');
+var getMockList = require('./assets/get_mock_list');
+var getRequestOpts = require('./assets/get_image_request_options');
+var getImagePaths = require('./assets/get_image_paths');
 
 // packages inside the image server docker
-var request = require('request');
 var test = require('tape');
+var request = require('request');
 var gm = require('gm');
 
-var TOLERANCE = 1e-6;    // pixel comparison tolerance
-var BASE_TIMEOUT = 500;  // base timeout time
-var BATCH_SIZE = 5;      // size of each test 'batch'
-var running = 0;         // number of tests currently running
+// pixel comparison tolerance
+var TOLERANCE = 1e-6;
+
+// wait time between each test batch
+var BATCH_WAIT = 500;
+
+// number of tests in each test batch
+var BATCH_SIZE = 5;
+
+// wait time between each test in test queue
+var QUEUE_WAIT = 10;
+
+/**
+ *  Image pixel comparison test script.
+ *
+ *  Called by `tasks/test_image.sh in `npm run test-image`.
+ *
+ *  CLI arguments:
+ *
+ *  1. 'pattern' : glob determining which mock(s) are to be tested
+ *  2. --queue : if sent, the image will be run in queue instead of in batch.
+ *      Makes the test run significantly longer, but is recommended on weak hardware.
+ *
+ *  Examples:
+ *
+ *  Run all tests in batch:
+ *
+ *      npm run test-image
+ *
+ *  Run the 'contour_nolines' test:
+ *
+ *      npm run test-image -- contour_nolines
+ *
+ *  Run all gl3d image test in queue:
+ *
+ *      npm run test-image -- gl3d_* --queue
+ */
+
+var pattern = process.argv[2];
+var mockList = getMockList(pattern);
+var isInQueue = (process.argv[3] === '--queue');
+
+if(mockList.length === 0) {
+    throw new Error('No mocks found with pattern ' + pattern);
+}
 
-var touch = function(fileName) {
-    fs.closeSync(fs.openSync(fileName, 'w'));
-};
+mockList = mockList.filter(untestableFilter);
 
+if(mockList.length === 0) {
+    throw new Error('All mocks found with pattern ' + pattern + ' are currently untestable');
+}
 
-// make artifact folders
-if(!fs.existsSync(constants.pathToTestImagesDiff)) {
-    fs.mkdirSync(constants.pathToTestImagesDiff);
+// main
+if(isInQueue) {
+    runInQueue(mockList);
 }
-if(!fs.existsSync(constants.pathToTestImages)) {
-    fs.mkdirSync(constants.pathToTestImages);
+else {
+    runInBatch(mockList);
 }
 
-var userFileName = process.argv[2];
+/* Test cases:
+ *
+ * - font-wishlist
+ * - all gl2d
+ * - all mapbox
+ *
+ * don't behave consistently from run-to-run and/or
+ * machine-to-machine; skip over them for now.
+ *
+ */
+function untestableFilter(mockName) {
```

Comment:
It would be useful for sure. But ideally, this filter wouldn't be there in the first place. So I'd vote for the solution of least-surface-area i.e. this one :arrow_double_up: .